esmvaltool/cmorizers/data/formatters/datasets/esacci_cloud.py

```diff
@@ -0,0 +1,404 @@
+
+    if short_name == "clwvi":
+        logger.info("Adding lwp and clivi")
+        cube_lwp = iris.load_cube(ifile, NameConstraint(var_name="lwp_allsky"))
+        monthly_cube.data = monthly_cube.core_data() + cube_lwp.core_data()
+
+    if monthly_cube is None:
+        logger.warning("Cube could not be loaded for file '%s'", ifile)
+        return  # Skip this file and move to the next
+
+    monthly_cube.attributes.clear()
+    monthly_cube.coord("time").long_name = "time"
+
+    # Fix coordinates
+    monthly_cube = utils.fix_coords(monthly_cube)
+    # Fix data type
+    utils.fix_dtype(monthly_cube)
+    # Fix metadata
+    utils.fix_var_metadata(monthly_cube, cmor_info)
+
+    # Add the cube to the list
+    if any(
+        sat_am in ifile
+        for sat_am in (
+            "AVHRR_NOAA-12",
+            "AVHRR_NOAA-15",
+            "AVHRR_NOAA-17",
+            "AVHRR_METOPA",
+        )
+    ):
+        cubes_am.append(monthly_cube)
+    elif any(
+        sat_pm in ifile
+        for sat_pm in (
+            "AVHRR_NOAA-7",
+            "AVHRR_NOAA-9",
+            "AVHRR_NOAA-11",
+            "AVHRR_NOAA-14",
+            "AVHRR_NOAA-16",
+            "AVHRR_NOAA-18",
+            "AVHRR_NOAA-19",
+        )
+    ):
+        cubes_pm.append(monthly_cube)
+    else:
+        logger.error("The file %s is not assigned to AM or PM", ifile)
+
+
+def _extract_variable_daily(
+    short_name, var, cfg, in_dir, out_dir, start_date, end_date
+):
+    """Extract daily variable."""
+    cmor_info = cfg["cmor_table"].get_variable(var["mip"], short_name)
+
+    if not start_date:
+        start_date = datetime(cfg["start_year_daily"], 1, 1)
+    if not end_date:
+        end_date = datetime(cfg["end_year_daily"], 12, 31)
+
+    for year in range(start_date.year, end_date.year + 1):
+        cubes = iris.cube.CubeList()
+        cubes_day = iris.cube.CubeList()
+
+        for month in range(1, 13):
+            num_days = monthrange(year, month)[1]
+            for iday in range(1, num_days + 1):
+                filelist = glob.glob(
+                    os.path.join(
+                        in_dir,
+                        f"{year}{month:02}" + f"{iday:02}" + var["file"],
+                    )
+                )
+
+                if filelist:
+                    for inum, ifile in enumerate(filelist):
+                        try:
+                            _process_daily_file(
+                                ifile,
+                                inum,
+                                short_name,
+                                var,
+                                cmor_info,
+                                cubes,
+                                cubes_day,
+                            )
+                        except Exception as e:
+                            logger.error(
+                                "Error processing file '%s': %s", ifile, e
+                            )
```

Comment:
We also want a clean exception here if something fails, otherwise you get a final cube with missing data and you won't know why